### DISCUSSION TITLE
Release 5.1.2

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1204] [Announcement]
* We now test against, and package with, Python 3.9.6.

  **As a consequence of this we no longer support Windows 7.
  This is due to [Python 3.9.x itself not supporting Windows 7](https://www.python.org/downloads/windows/).
  The application (both EDMarketConnector.exe and EDMC.exe) will crash on startup due to a missing DLL.**

  This should have no other impact on users or plugin developers, other than the latter now being free to use features that were introduced since the Python 3.7 series.

  Developers can check the contents of the `.python-version` file in the source (it's not distributed with the Windows installer) for the currently used version in a given branch.

Release 5.1.2
===

* A Journal event change in EDO Update 6 will have caused some translated suit names to not be properly mapped to their sane versions.  This change has now been addressed and suit names should always come out as intended in the EDMarketConnector.exe UI.

* There is a new command-line argument to cause all Frontier Authorisation to be forgotten:  `EDMarketConnector.exe --forget-frontier-auth`.

* Situations where Frontier CAPI data doesn't agree on the location we have tracked from Journal events will now log more useful information.

Bug Fixes
---

* The code should now be robust against the case of any Journal event name changing.

Plugin Developers
---

* We now store `GameLanguage`, `GameVersion` and `GameBuild` in the `state` passed to `journal_entry()` from the `LoadGame` event.

* Various suit data, i.e. class and mods, is now stored from relevant Journal events, rather than only being available from CAPI data.  In general, we now consider the Journal to be the canonical source of suit data, with CAPI only as a backup.

* Backpack contents should now track correctly if using the 'Resupply' option available on the ship boarding menu.

* We now cache the main application version when first determined, so that subsequent references to `config.appversion()` won't cause extra log spam (which was possible when, e.g. having a git command but using non-git source).

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.1.2'>Release 5.1.2</a>.</em>

**Comment by Athanasius** [comment]
I've not yet set 5.1.2 to be visible to the auto-updater, that might not happen until tomorrow.

**Reply by Athanasius** [reply]
I performed that update over an hour ago, so auto-updates should be happening for everyone today.

**Comment by Athanasius** [comment]
Oh, I totally forgot to call out that I also updated translations in 5.1.2, so thanks again to all the translators!

**Comment by kateshepard** [comment]
My virus scanner found a trojan in the .msi file... anyone else experiencing this?

**Reply by Athanasius** [reply]
To pre-empt the now seemingly inevitable anti-virus false positive reports I'd already uploaded the MSI and two EXE files for the forthcoming 5.1.2 release to VirtusTotal:

5.1.2 installer: https://www.virustotal.com/gui/file/a5c2db9e4a01f293b7e95f4178530e4e7d374040a62ef91434b7759de4cb8bd2/detection
EDMarketConnector.exe: https://www.virustotal.com/gui/file/ef2f7b9bb984582fa6ebeade40efb8734b5c575309dfa04207ec4f71bf64052b/detection
EDMC.exe: https://www.virustotal.com/gui/file/04de79a6f6d1294323f5aba9cc916bb0828f14b3661a6f5647d78ea66ebb6e5f/detection

Some detections showed up, but not in any major AV.  This will be, yet again, some of them detecting a false positive due to us *and malware authors* using py2exe to build python source code into executables.

If you're truly worried then you can check the source code (which you're trusting anyway) and run from it, rather than the installer.  See our Wiki for details.

**Comment by kateshepard** [comment]
Hi Athanasius,

Thank you for reaching out directly to me. I just added the app .exe file to the Safe List.

 

Kate
